app/src/main/java/com/pokedex/mdb/pokedex/ListActivity.java

```diff
@@ -0,0 +1,230 @@
+package com.pokedex.mdb.pokedex;
+
+/**
+ * Created by Aayush on 9/28/2016.
+ */
+
+        import android.app.SearchManager;
+        import android.content.Context;
+        import android.support.v7.app.AppCompatActivity;
+        import android.os.Bundle;
+        import android.support.v7.widget.GridLayoutManager;
+        import android.support.v7.widget.LinearLayoutManager;
+        import android.support.v7.widget.RecyclerView;
+        import android.support.v7.widget.SearchView;
+        import android.view.Menu;
+        import android.view.MenuInflater;
+        import android.view.MenuItem;
+
+        import java.util.ArrayList;
+
+public class ListActivity extends AppCompatActivity {
+
+    private RecyclerView recycleboy;
+    private boolean isLinear = false;
+    private Pokedapter pokedapter;
+    private ArrayList<Pokedex.Pokemon> pokemons;
+    private ArrayList<Pokedex.Pokemon> searchingcache;
+    private ArrayList<Pokedex.Pokemon> filteringcache;
+
+    private ArrayList<Pokedex.Pokemon> lowHP;

```

Comment:
I would store indexes instead of cloning every pokemon in each category. THINK ABOUT THEIR FEELINGS TOO :(
Also wastes a lot of spaces and makes object modification impossible, if you wanted to, say, change an entry.